### DISCUSSION TITLE
Is this Vagrant configuration possible in lima?

### DISCUSSION THREAD

**Discussion by blissdev** [lima-vm/lima#480] [Q&A]
I'm looking for possible solutions to replacing our Vagrant/Virtualbox setup on ARM64 machines. Does this networking setup seem viable in lima?

```ruby
Vagrant.configure("2") do |config|
    boxes = ["db", "test.example.com", "web", "cache", "config", "sftp", "bastion"]
    boxes.each_with_index do |name, i|
        config.vm.define name do |new_box|
            new_box.vm.box = "centos/7"
            new_box.vm.hostname = name
            new_box.vm.network "private_network", ip: "192.168.56.#{10+i}"
    end
end
```

I guess the main thing I'm looking for is that the boxes can talk to each other via the network. If centOS isn't available, could just use Fedora.

I noticed these PRs:

https://github.com/lima-vm/lima/discussions/408#discussioncomment-1634598
https://github.com/lima-vm/vde_vmnet/issues/19#issuecomment-950109235

Are those the direction that I would need to go?

**Comment by jandubois** [comment]
Yes, you should be able to create several VMs with IP addresses on a shared network. Just add

```yaml
networks:
- lima: shared
```

to your `lima.yaml` file, and `limactl` should take care of everything. You will need to use `limactl sudoers` to create a sudoers file unless you have passwordless sudo enabled for your current user.

You will also not be able to pick the IP addresses; they will be assigned by DHCP, but should remain stable as long as you don't rename the instance, or move your `LIMA_HOME` directory.

**Reply by blissdev** [reply]
Thanks @jandubois! How do I make the multiple VMs? Do I just have to call `limactl start instance-of-7.yaml` all 7 times? Or can I orchestrate within?

**Reply by jandubois** [reply]
Yes, you have to run `limactl start` for each instance. You can write a shell script... 😄

**Reply by blissdev** [reply]
Makes sense. One last question, how do I figure out the assigned IP the created instance?

**Reply by jandubois** [reply]
You have to look it up yourself, e.g. `limactl shell $INSTANCE ip addr show lima0`, using the default name for the first defined lima network interface.

**Reply by blissdev** [reply]
Thank you!
